src/Datadog.Trace/Tracer.cs

```diff
@@ -353,6 +355,18 @@ void IDatadogTracer.Write(Span[] trace)
             }
         }
 
+        /// <summary>
+        /// Reports the detected version of the agent
+        /// </summary>
+        /// <param name="version">Version of the agent</param>
+        void IDatadogTracer.ReportAgentVersion(string version)
+        {
+            if (ShouldLogPartialFlushWarning(version))
+            {
+                Log.Warning("DATADOG TRACER DIAGNOSTICS - Partial flush should only be enabled with agent 7.26.0+ (detected version: {version})", version);
```

Comment:
If this is detected, do we fallback to non-partial flush?

Reply:
I can't reliably detect the agent, because 7.26.0 has no version check (even though it's fine to use with partial flush), or simply because the agent may not be responsive at the moment the app is started. Because of that, I think it's better to just log a warning. I'm afraid false negatives could cause much pain.

Reply:
Isn't this going to say `(detected version: )` with older Agent versions?